src/psyclone/psyir/nodes/acc_directives.py

```diff
@@ -89,11 +88,29 @@ def validate_global_constraints(self):
         data_nodes = self.walk((PSyDataNode, CodeBlock))
         if data_nodes:
             raise GenerationError(
-                "Cannot include CodeBlocks or calls to PSyData routines within"
-                " OpenACC regions but found {0} within a region enclosed "
-                "by an '{1}'".format(
-                    [type(node).__name__ for node in data_nodes],
-                    type(self).__name__))
+                f"Cannot include CodeBlocks or calls to PSyData routines "
+                f"within OpenACC regions but found "
+                f"{[type(node).__name__ for node in data_nodes]} within a "
+                f"region enclosed by an '{type(self).__name__}'")
+
+    @property
+    def ref_set(self):
```

Comment:
I think a better name for this would be `kernel_references`.

Reply:
Sure!

Reply:
Oh yes, that comment is not quite accurate - this is dealing with more than just fields (e.g. the GOcean grid properties).